csvGenerator.ipynb

```diff
@@ -0,0 +1,195 @@
+    "\"\"\"\n",
+    "\n",
+    "def write_to_csv(csv_name:str,item:list,flag:int):\n",
+    "    with open(csv_name, 'a', newline='') as file:\n",
+    "        writer = csv.writer(file)\n",
+    "        if flag==0:\n",
+    "            fields = list(item.keys())\n",
+    "            writer.writerow(list(fields))\n",
+    "        writer.writerow(list(item.values()))\n",
+    "        \n",
+    "# Add the item to csv_dict that contains all the collection items(with metadata and downloaded file path)\n",
+    "\n",
+    "\"\"\"\n",
+    "adds items to a dictionary object; csv_dict \n",
+    "with the collection items unique key as the key for the dictionary\n",
+    "and the item dictionary as the value for the respective key, \n",
+    "that contains our fields and value/metadata for a\n",
+    "given collection item that we write to the csv file. \n",
+    "It also adds a path property that indicates the location of the downloades files(if any)\n",
+    "\"\"\"\n",
+    "\n",
+    "def add_to_csv(csv_name:str,folder_name:str, items:list,csv_dict:dict,flag:int):\n",
+    "    for item in items[\"items\"]:\n",
+    "        if item[\"key\"] in csv_dict:\n",
+    "            continue\n",
+    "        item[\"path\"] = \"\"\n",
+    "        if item[\"gilesUploads\"]:\n",
+    "            giles_ids = get_set(item[\"gilesUploads\"])\n",
+    "            if giles_ids:\n",
+    "                item[\"path\"] = download_file(folder_name, giles_ids, auth_object.access_token)\n",
+    "        csv_dict[item[\"key\"]] = item\n",
+    "        write_to_csv(csv_name,item,flag)\n",
+    "        flag = 1\n",
+    "    return flag"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "id": "229c7e47",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "#downloads and generate a csv file containing all the group items information\n",
+    "def process_groups(csv_name:str,folder_path:str,groups:list, connector, max_size:int):\n",
+    "    csv_dict = {}\n",
+    "    flag = 0\n",
+    "    for group in groups:\n",
+    "        group_id = group[\"id\"]\n",
+    "        collections = connector.get_collections(group_id)\n",
+    "        \n",
+    "        for collection in collections[\"collections\"]:\n",
+    "            num_pages = math.ceil(collection[\"numberOfItems\"] / max_size)\n",
+    "\n",
+    "            if num_pages == 1:\n",
+    "                items = connector.get_collection_items_pg(group_id, collection[\"key\"], 1)\n",
+    "                flag = add_to_csv(csv_name, folder_path, items, csv_dict, flag)\n",
+    "            elif num_pages > 1:\n",
+    "                for page in range(num_pages):\n",
+    "                    items = connector.get_collection_items_pg(group_id, collection[\"key\"], page + 1)\n",
+    "                    flag = add_to_csv(csv_name, folder_path, items, csv_dict, flag)\n",
```

Comment:
I made this comment before I believe, you can iterate over a range of 1 I believe